machine-runner/src/runner/runner.ts

```diff
@@ -83,23 +84,66 @@ export type MachineRunner = {
    * }
    * machine.isDestroyed() // returns false
    */
-  noAutoDestroy: () => MachineRunnerIterableIterator
-} & MachineRunnerIterableIterator
+  noAutoDestroy: () => MachineRunnerIterableIterator<SwarmProtocolName, MachineName>
+} & MachineRunnerIterableIterator<SwarmProtocolName, MachineName>
 
 export namespace MachineRunner {
-  type AllEventMap = MachineEmitterEventMap
-  export type EventListener<Key extends keyof AllEventMap> = AllEventMap[Key]
+  /**
+   * The widest type of MachineRunner. Any other MachineRunner extends this type
+   */
+  export type Any = MachineRunner<string, string>
+
+  /**
+   * Extract MachineRunner event emitter map type from a MachineRunner
+   * @example
+   * const machineRunner = createMachineRunner(actyx, where, Passenger.Initial, void 0);
+   *
+   * type EventMap = MachineRunner.EventMapOf<typeof machineRunner>
+   * type OnChange = EventMap['change']
+   *
+   * const onChange: EventMap['change'] = () =>
+   *  console.log(label, 'state after caughtUp', utils.deepCopy(machine.get()))
+   * machine.events.on('change', onChange)
+   *
+   * // later
+   * machine.events.off('change', onChange)
+   */
+  export type EventMapOf<M extends MachineRunner<any, any>> = M extends MachineRunner<
+    infer S,
+    infer N
+  >
+    ? MachineEmitterEventMap<S, N>
+    : never
+
+  /**
+   * Extract MachineRunner type from SwarmProtocol or Machine
+   * @example
+   * const HangarBay = SwarmProtocol.make('HangarBay', ['hangar-bay'], Events.all)
+   * const Door = HangarBay.makeMachine('door')
+   * const Initial = Door.designEmpty().finish()
+   *
+   * // refers to any MachineRunner derived from HangarBay protocol
+   * type ThisMachineRunner = MachineRunner.Of<typeof HangarBay>
+   *
+   * // refers to any MachineRunner derived from HangarBay protocol and Door machine
+   * type ThisMachineRunner = MachineRunner.Of<typeof Door>
+   */
+  export type Of<S extends SwarmProtocol<any, any, any> | Machine<any, any, any>> =
+    S extends SwarmProtocol<infer S, any, any>
+      ? MachineRunner<S, string>
```

Comment:
What are the users going to use this case for?

Reply:
It is for when a user needs a MachineRunner type that is for a particular swarm and not for a particular Machine. e.g. when a user wants to author a library that depends on a certain protocol, which includes functions catering all machine runners that the protocol produces. It is used in `dev-example`. 

```
type ThisMachineRunner = MachineRunner.Of<typeof protocol>
```